src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,207 @@
+package duke;
+
+import duke.tasktypes.Deadlines;
+import duke.tasktypes.Events;
+import duke.tasktypes.ToDo;
+import duke.tasktypes.Task;
+
+/**
+ * Class used to interpret the user's input for Duke chatbot.
+ */
+public class Parser {
+    
+    public Parser() {}
+
+    /**
+     * Function to redirect user's input to the relevant functions.
+     * @param input String representation of the user's input.
+     * @param listOfTasks The list of tasks which has been loaded into the Duke chatbot instance.
+     * @throws DukeExceptions if the user inputs an invalid command.
+     */
+    public void readInput(String input, TaskList listOfTasks) throws DukeExceptions {
+        
+        try {
+            if (input.startsWith("todo")) {
+                handleToDoTask(input, listOfTasks);
+                return;
+            }
+
+            if (input.startsWith("find")) {
+                handleFindTask(input, listOfTasks);
+                return;
+            }
+
+            if (input.startsWith("deadline")) {
+                handleDeadlineTask(input, listOfTasks);
+                return;
+            }
+
+            if (input.startsWith("event")) {
+                handleEventTask(input, listOfTasks);
+                return;
+            }
+
+            if (input.equals("list")) {
+                handleList(listOfTasks);
+                return;
+            }
+
+            if (input.startsWith("delete")) {
+                handleDelete(input, listOfTasks);
+                return;
+            }
+
+            if (input.startsWith("checkdue")) {
+                handleCheckDue(input, listOfTasks);
+                return;
+            }
+
+            if (input.startsWith("mark")) {
+                handleMark(input, listOfTasks);
+                return;
+            }
+
+            if (input.startsWith("unmark")) {
+                handleUnmark(input, listOfTasks);
+                return;
+            }
```

Comment:
One way to use 'startsWith' together with a switch statement is to use the split method.